gtsam/linear/GaussianConditional.cpp

```diff
@@ -192,7 +224,44 @@ namespace gtsam {
     }
   }
 
-  /* ************************************************************************* */
+  /* ************************************************************************ */
+  VectorValues GaussianConditional::sample(const VectorValues& parentsValues,
+                                           std::mt19937_64* rng) const {
+    if (nrFrontals() != 1) {
+      throw std::invalid_argument(
+          "GaussianConditional::sample can only be called on single variable "
+          "conditionals");
+    }
+    if (!model_) {
+      throw std::invalid_argument(
+          "GaussianConditional::sample can only be called if a diagonal noise "
```

Comment:
This says diagonal noise model, but the check is just for a model existing rather than a diagonal one. So perhaps update the error message?

Reply:
GaussianConditional inherits from JacobianFactor which only accepts diagonal models.